### PR TITLE
Allow multiple Cluster JMX MBeans in the same JVM

### PR DESCRIPTION
**Pull request by richardimaoka**
Refs #18872 

I didn't want to change the default behavior which impacts existing users ... so add a new flag in config.

`akka.cluster.jmx.multi-mbeans-in-same-jvm = on` in config allows you register multiple Cluster JMX MBeans in the same JVM. It is `= off` by default, which preserves the old behavior. 

`jmx-client/akka-cluster` is also updated. When `akka.cluster.jmx.multi-mbeans-in-same-jvm = off` on the server (cluster) side, no change in behavior. If that is `= on`, you need to pass the `-p` optional parameter.

### akka.cluster.jmx.multi-mbeans-in-same-jvm = off, on the server (cluster) side

* akka-cluster works same way as before

```
$ ./akka-cluster 127.0.0.1 9999 member-status
This jmx-client/akka-cluster tool is deprecated use curl and https://github.com/akka/akka-cluster-management instead, since 2.5.0.
Querying member status for 127.0.0.1
Up
```

### akka.cluster.jmx.multi-mbeans-in-same-jvm = on, on the server (cluster) side

* it doesn't work if you are missing -p

```
$ ./akka-cluster 127.0.0.1 9999 member-status
This jmx-client/akka-cluster tool is deprecated use curl and https://github.com/akka/akka-cluster-management instead, since 2.5.0.
Akka cluster MBean is not available on 127.0.0.1:9999 with MBean name 'akka:type=Cluster'
```

* -p needs to be passed

```
$ ./akka-cluster 127.0.0.1 9999 -p 2552 down akka.tcp://ClusterSystem@127.0.0.1:2551
This jmx-client/akka-cluster tool is deprecated use curl and https://github.com/akka/akka-cluster-management instead, since 2.5.0.
Marking akka.tcp://ClusterSystem@127.0.0.1:2551 as DOWN
```

* port 2551 is expected unavailable, as it was made DOWN above

```
$ ./akka-cluster 127.0.0.1 9999 -p 2551 member-status
This jmx-client/akka-cluster tool is deprecated use curl and https://github.com/akka/akka-cluster-management instead, since 2.5.0.
Akka cluster MBean is not available on 127.0.0.1:9999 with MBean name 'akka:type=Cluster,port=2551'
```

* but port 2552 is still available

```
$ ./akka-cluster 127.0.0.1 9999 -p 2552 cluster-status
This jmx-client/akka-cluster tool is deprecated use curl and https://github.com/akka/akka-cluster-management instead, since 2.5.0.
{
  "members": [
    {
      "address": "akka.tcp://ClusterSystem@127.0.0.1:2552",
      "roles": [],
      "status": "Up"
    },
    {
      "address": "akka.tcp://ClusterSystem@127.0.0.1:57642",
      "roles": [],
      "status": "Up"
    }
  ],
  "self-address": "akka.tcp://ClusterSystem@127.0.0.1:2552",
  "unreachable": []
}
```

### CODE REVIEW

File: akka-cluster/src/main/scala/akka/cluster/ClusterSettings.scala [hunk 1]
```diff
@@ -105,6 +105,7 @@ final class ClusterSettings(val config: Config, val systemName: String) {
   }
   val RunCoordinatedShutdownWhenDown: Boolean = cc.getBoolean("run-coordinated-shutdown-when-down")
   val JmxEnabled: Boolean = cc.getBoolean("jmx.enabled")
+  val JmxMultiMbeansInSameEnabled: Boolean = cc.getBoolean("jmx.multi-mbeans-in-same-jvm ")
```
Comment (2m) [inline akka-cluster/src/main/scala/akka/cluster/ClusterSettings.scala]: Remove superfluous space at the end.

**Review by 2m — CHANGES_REQUESTED** [review]
In general this looks great! As this adds support for multiple clusters running on one JVM it should also handle the case where separate clusters are listening on the same port but different hostname. This could be the case on nodes with multiple NICs.

File: akka-cluster/src/main/scala/akka/cluster/ClusterJmx.scala [hunk 2]
```diff
@@ -199,7 +204,15 @@ private[akka] class ClusterJmx(cluster: Cluster, log: LoggingAdapter) {
...
       logInfo("Registered cluster JMX MBean [{}]", clusterMBeanName)
     } catch {
-      case e: InstanceAlreadyExistsException ⇒ // ignore - we are running multiple cluster nodes in the same JVM (probably for testing)
+      case e: InstanceAlreadyExistsException ⇒ {
+        if (cluster.settings.JmxMultiMbeansInSameEnabled) {
+          log.error(e, s"Failed to register Cluster JMX MBean with name=$clusterMBeanName")
+        } else {
+          log.warning(
+            s"Could not register Cluster JMX MBean with name=$clusterMBeanName as it is already registered. " +
+              "If you are running multiple clusters in the same JVM, set 'akka.cluster.jmx.multi-mbeans-in-same-jvm = on' in config")
```
Comment (patriknw) [inline akka-cluster/src/main/scala/akka/cluster/ClusterJmx.scala]: good

File: akka-cluster/src/main/scala/akka/cluster/ClusterSettings.scala (hunk 1 above)
Comment (patriknw) [inline akka-cluster/src/main/scala/akka/cluster/ClusterSettings.scala]: yes, better fix this

**Review by patriknw — CHANGES_REQUESTED** [review]
Looking very good. I think the only needed change is the whitespace. @2m has a valid point with multiple NICs, but in practice I guess that combination is rare and not worth spending time on.

File: akka-cluster/src/main/scala/akka/cluster/ClusterSettings.scala (hunk 1 above)
Comment (richard-imaoka) [inline akka-cluster/src/main/scala/akka/cluster/ClusterSettings.scala]: Hi yes, sorry I overlooked this but now fixed it in the latest commit.

**Review by 2m — APPROVED** [review]

**Review by patriknw — APPROVED** [review]
LGTM

**Comment by patriknw** [comment]
Great work @richard-imaoka

**Pull request merged**